packages/EasyTest/src/Coverage/Factory/CoverageResolverFactory.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EonX\EasyTest\Coverage\Factory;
+
+use EonX\EasyTest\Coverage\Resolvers\CloverCoverageResolver;
+use EonX\EasyTest\Coverage\Resolvers\TextCoverageResolver;
+use EonX\EasyTest\Interfaces\CoverageResolverFactoryInterface;
+use EonX\EasyTest\Interfaces\CoverageResolverInterface;
+use InvalidArgumentException;
+
+final class CoverageResolverFactory implements CoverageResolverFactoryInterface
+{
+    private const CLOVER_EXTENSION = 'clover';
+
+    private const TXT_EXTENSION = 'txt';
+
+    public function create(string $filePath): CoverageResolverInterface
+    {
+        $ext = pathinfo($filePath, PATHINFO_EXTENSION);
+        switch ($ext) {
+            case self::CLOVER_EXTENSION:
+                $resolver = new CloverCoverageResolver();
```

Comment:
I believe we must use these resolvers as services. I.e. to fetch them from the container. What about to use the same solution as we use in Payments in Locator-classes instead of factories?

Reply:
done